### DISCUSSION TITLE
Feature Request: Add  HAS_BOM  option  in csv reader

### DISCUSSION THREAD

**Discussion by douenergy** [duckdb/duckdb#6727] [Ideas]
"The BOM (byte order mark) is a particular usage of the special Unicode character, U+FEFF BYTE ORDER MARK, whose appearance as a magic number at the start of a text stream can signal several things to a program reading the text"

[people.csv](https://github.com/duckdb/duckdb/files/10978294/people.csv)

```sql 
FROM 'people.csv' ; -- works fine
```
```sql
SELECT id FROM 'people.csv'; -- Error: Binder Error: Referenced column "id" not found in FROM clause!
```
The problem is that there is a "BOM" character before **id**.

<img width="342" alt="image" src="https://user-images.githubusercontent.com/103009868/225273348-8dbf28b2-3f69-4926-8f8b-f6d88fea4567.png">

Pandas can solve this by specifying the engine.

```python
import pandas as pd 
people = pd.read_csv('./people.csv', engine='python')
print(people['id'])
```

https://en.wikipedia.org/wiki/Byte_order_mark

**Comment by douenergy** [comment]
https://github.com/duckdb/duckdb/pull/3014 
seems not working

**Comment by Mytherin** [comment]
The BOM in this file is not actually valid. The BOM has to be the first three bytes of the file. The first byte of the file is the quote instead. If we modify the file so the BOM is actually at the start this works as expected:

```csv
"id","name"
"1","Alice"
"2","Bob"
```

```sql
SELECT id FROM '~/Downloads/people.csv';
┌───────┐
│  id   │
│ int64 │
├───────┤
│     1 │
│     2 │
└───────┘
```

You can also use `normalize_names` to drop special characters from the column names:

```sql
SELECT id FROM read_csv_auto('~/Downloads/people.csv', normalize_names=True);
┌───────┐
│  id   │
│ int64 │
├───────┤
│     1 │
│     2 │
└───────┘
```

**Comment by douenergy** [comment]
Thanks ! normalize_names works  perfectly

**Comment by jimdickson007** [comment]
I think the behaviour of how duckdb (_read_csv_) handles BOM should be explicitly documented.
I searched for a long time before I stumbled on _normalize_names_.
Actually appears to drop BOM with default normalize_names = false
